fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/FullUnorderedScanExpression.java

```diff
@@ -134,7 +143,15 @@ public String toString() {
     @Nonnull
     @Override
     public Iterable<MatchInfo> subsumedBy(@Nonnull final RelationalExpression candidateExpression, @Nonnull final AliasMap aliasMap, @Nonnull final IdentityBiMap<Quantifier, PartialMatch> partialMatchMap) {
-        return exactlySubsumedBy(candidateExpression, aliasMap, partialMatchMap);
+        if (getClass() != candidateExpression.getClass()) {
+            return ImmutableList.of();
+        }
+        // if query doesn't contain any index, or contains candidate's index, the query should be exactly subsumed by the candidate
+        if (getIndexSet().size() == 0 || getIndexSet().containsAll(((FullUnorderedScanExpression)candidateExpression).getIndexSet())) {
```

Comment:
Please use `.isEmpty()` instead (IntelliJ gives you a warning actually). Also, I think instead of defaulting `isEmpty()` ==> no hinting, why not just setting all available index names when the fuse is created and no hints are specified.

Reply:
Good idea! will set all available index names, it is clearer.